### PR TITLE
[RW-12567][risk=no]  [UI] Feature workspace tab feature workspaces from new getFeatureWorkspace endpoint

### PR DESCRIPTION
**Pull request by NehaBroad**
https://github.com/user-attachments/assets/85407962-e787-43ce-90f0-9b1362a78d6c

---
**PR checklist**

- [ ] I have included an issue ID or "no ticket" in the PR title as outlined in [CONTRIBUTING.md](https://github.com/all-of-us/workbench/blob/main/.github/CONTRIBUTING.md).
- [ ] I have included a risk tag of the form `[risk=no|low|moderate|severe]` in the PR title as outlined in [CONTRIBUTING.md](https://github.com/all-of-us/workbench/blob/main/.github/CONTRIBUTING.md).
- [ ] I have manually tested this change and my testing process is described above.
- [ ] This change includes appropriate automated tests, and I have documented any behavior that cannot be tested with code.
- [ ] I have added explanatory comments where the logic is not obvious.
- One or more of the following is true:
  - [ ] This change is intended to complete a JIRA story, so I have checked that all AC are met for that story.
  - [ ] This change fixes a bug, so I have ensured the steps to reproduce are in the Jira ticket or provided above.
  - [ ] This change impacts deployment safety (e.g. removing/altering APIs which are in use), so I have documented the impacts in the description.
  - [ ] This change includes a new feature flag, so I have created and linked new JIRA tickets to (a) turn on the feature flag and (b) remove it later.
  - [ ] This change modifies the UI, so I have taken screenshots or recordings of the new behavior and notified the PO and UX designer in [Slack](https://pmi-engteam.slack.com/archives/C02MWP2RN5P).
  - [ ] This change modifies API behavior, so I have run the relevant E2E tests locally because API changes are not covered by our PR checks.
  - [ ] None of the above apply to this change.

### CODE REVIEW

File: ui/src/app/pages/workspace/workspace-card.tsx [hunk 1]
```diff
@@ -284,6 +286,11 @@ export const WorkspaceCard = fp.flow(withNavigation)(
                     <div style={{ fontSize: 12 }}>
                       Last Changed: {displayDate(workspace.lastModifiedTime)}
                     </div>
+                    {workspaceLibrary && (
```
Comment (NehaBroad) [inline ui/src/app/pages/workspace/workspace-card.tsx:289]: Good to show who created the workspace in featured workspace as there could be multiple owners sharing workspace with same name

File: ui/src/app/pages/workspace/featured-workspaces.tsx [hunk 2]
```diff
@@ -0,0 +1,302 @@
+import * as React from 'react';
```
Comment (NehaBroad) [inline ui/src/app/pages/workspace/featured-workspaces.tsx:1]: Basically copied from workspace-library with minor tweaks and changing the source to getFeaturedWorkspacesByCategory.

File: ui/src/app/pages/workspace/featured-workspaces.tsx [hunk 3]
```diff
@@ -0,0 +1,302 @@
...
+      are aimed at demonstrating the quality, utility, and diversity of <AoU />{' '}
+      data by replicating previously published studies. You can open the
+      workspaces to view them or “duplicate” the workspaces to edit and execute
+      them.
+    </div>,
+    demonstration,
+    FeaturedWorkspaceCategory.DEMO_PROJECTS
+  ),
+  COMMUNITY: createTab(
+    'Community Projects',
```
Comment (jmthibault79) [inline ui/src/app/pages/workspace/featured-workspaces.tsx:130]: why not "Community Workspaces" ?

File: ui/src/app/pages/workspace/featured-workspaces.tsx [hunk 4]
```diff
@@ -0,0 +1,302 @@
...
+    <div>
+      Community workspaces are curated and maintained by researchers using the
+      Research Workbench and are intended for use by Research Workbench
```
Comment (jmthibault79) [inline ui/src/app/pages/workspace/featured-workspaces.tsx:133]: add "the"

Comment (jmthibault79) [reply ui/src/app/pages/workspace/featured-workspaces.tsx:133]: also: it should be Research**er** Workbench

**Review by jmthibault79 — CHANGES_REQUESTED** [review]
Looks good overall, but I have a few comments and requests

File: ui/src/app/pages/workspace/featured-workspaces.tsx [hunk 5]
```diff
@@ -0,0 +1,314 @@
...
+});
+
+interface LibraryTabProps {
+  title: string;
+  description: ReactNode;
+  icon: string;
+  category: FeaturedWorkspaceCategory;
+}
+
+const createTab = (title, description, icon, category): LibraryTabProps => ({
```
Comment (jmthibault79) [inline ui/src/app/pages/workspace/featured-workspaces.tsx:86]: I'm not sure this indirection serves a purpose any more.  What do you think about defining the tabs inline instead? That seems clearer to me.

Like:
```
PHENOTYPE_LIBRARY: {
    title: 'Phenotype Library',
    description: <div>
      These workspaces demonstrate how computable electronic phenotypes can be
      implemented within the <AoU /> dataset using examples of previously
      published phenotype algorithms. You can open the workspaces to view them
      or “duplicate” the workspaces to edit and execute the algorithms.
    </div>,
    icon: phenotypeLibrary,
    category: FeaturedWorkspaceCategory.PHENOTYPE_LIBRARY
  },
```

File: api/src/main/java/org/pmiops/workbench/api/FeaturedWorkspaceController.java [hunk 6]
```diff
@@ -37,4 +44,20 @@ public ResponseEntity<WorkspaceResponseListResponse> getFeaturedWorkspaces() {
           "Not implemented in this environment: combine the results of getFeaturedWorkspacesConfig() and getPublishedWorkspaces() to generate the list of featured workspaces.");
     }
   }
+
+  @Override
+  public ResponseEntity<WorkspaceResponseListResponse> getFeaturedWorkspacesByCategory(
```
[the remark this answers is not in the document]
Comment (jmthibault79) [reply api/src/main/java/org/pmiops/workbench/api/FeaturedWorkspaceController.java:49]: sounds good

**Review by jmthibault79 — COMMENTED** [review]
almost there - I want a change to the mapper however

File: ui/src/app/pages/workspace/featured-workspaces.tsx (hunk 5 above)
Comment (NehaBroad) [reply ui/src/app/pages/workspace/featured-workspaces.tsx]: Yeah that makes sense

File: ui/src/app/pages/workspace/workspace-card.tsx [hunk 7]
```diff
@@ -95,6 +95,7 @@ interface WorkspaceCardProps extends NavigationProps {
   reload(): Promise<void>;
   // non-CT users cannot click or see on CT workspaces.
   tierAccessDisabled?: boolean;
+  isOriginFeaturedWorkspace?: boolean;
```
[the remark this answers is not in the document]
Comment (NehaBroad) [reply ui/src/app/pages/workspace/workspace-card.tsx]: Updated

File: api/src/main/java/org/pmiops/workbench/utils/mappers/WorkspaceMapper.java [hunk 8]
```diff
@@ -84,6 +84,32 @@ default List<WorkspaceResponse> toApiWorkspaceResponseList(
...
+  @Mapping(target = "workspace.cdrVersionId", source = "dbWorkspace.cdrVersion")
+  @Mapping(
+      target = "workspace.accessTierShortName",
+      source = "dbWorkspace.cdrVersion.accessTier.shortName")
+  @Mapping(target = "workspace.googleProject", source = "dbWorkspace.googleProject")
+  @Mapping(
+      target = "accessLevel",
+      source = "accessLevel",
+      qualifiedByName = "fcToApiWorkspaceAccessLevel")
+  WorkspaceResponse toWorkspaceResponseWithAccess(
```
[the remark this answers is not in the document]
Comment (NehaBroad) [reply api/src/main/java/org/pmiops/workbench/utils/mappers/WorkspaceMapper.java]: I used the existing two mapper methods, let me know if that works

[the remark this answers is not in the document]
Comment (jmthibault79) [reply api/src/main/java/org/pmiops/workbench/utils/mappers/WorkspaceMapper.java]: LGTM!  Thanks.

File: ui/src/app/pages/workspace/featured-workspaces.tsx [hunk 9]
```diff
@@ -0,0 +1,315 @@
...
+  );
+};
+
+export const FeaturedWorkspaces = (props) => {
+  const [currentTab, setCurrentTab] = useState<LibraryTabProps>(
+    libraryTabs.TUTORIAL_WORKSPACES
+  );
+  const [errorText, setErrorText] = useState<string>('');
+
+  const [workspaceList, setWorkspaceList] = useState<Array<WorkspaceResponse>>(
```
Comment (jmthibault79) [inline ui/src/app/pages/workspace/featured-workspaces.tsx:175]: optional nitpick: anywhere you can use `Array<Whatever>` you can instead use `Whatever[]` -  it means the same thing.

File: ui/src/app/pages/workspace/featured-workspaces.spec.tsx [hunk 10]
```diff
@@ -0,0 +1,96 @@
...
+    component();
+    expect(
+      await screen.findByText('Researcher Workbench Workspace Library')
+    ).toBeInTheDocument();
+    featuredCategoryTabs.map((tabName) => {
+      expect(screen.getByRole('button', { name: tabName })).toBeInTheDocument();
+    });
+  });
+
+  it('Shows each workspaces on clicking Tab name', async () => {
```
Comment (jmthibault79) [inline ui/src/app/pages/workspace/featured-workspaces.spec.tsx:70]: "all workspaces" would be a better way to say this

**Review by jmthibault79 — APPROVED** [review]
Tested locally using a few combinations, including turning the feature flag off.  Works as expected in all situations.  Nice!

**Pull request merged**